### PR TITLE
Implement a requireJSDoc compatible rule with working tests

### PR DESCRIPTION
**Pull request by allthesignals**
Forked from the proposed pull request here: https://github.com/gajus/eslint-plugin-jsdoc/pull/188.

Original text from @thsmi: 
> Implements a requireJSDoc compatible rule as described in issue #162.
>
> The unit test file is ported from eslint and therefore MIT licensed for more details refer also to issue #162

### CODE REVIEW

**Comment by allthesignals** [comment]
Actually, the tests pass locally, so I'm not sure what's happening.

**Comment by allthesignals** [comment]
Should be good to go! The latest commit here is a cleaner approach using `git mv` and `git rm`.

File: src/rules/requireJsdoc.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
+//
```
Comment (gajus) [inline src/rules/requireJsdoc.js]: Unnecessary comment.

File: src/rules/requireJsdoc.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+// The requireJDDoc rule at eslint is deprecated and won't get any bugfixes anymore.
+// Thus this method implements a compatible requireJSDoc rule.
+//
+
+/* eslint-disable sort-keys */
```
Comment (gajus) [inline src/rules/requireJsdoc.js]: Unnecessary eslint-disable/ comment.

Sorted property style is enforced across the entire library. This rule implementation is not an exception.

**Comment by gajus** [comment]
Thank you for cleaning this up. I've added to style comments. Can you please address them? Otherwise, I am happy to jump in and amend the the PR in the next 24 hours.

**Comment by allthesignals** [comment]
@gajus I will be happy to - @thsmi, is this okay? I feel I am co-opting your original PR. Otherwise, will continue with cleanup.

**Comment by allthesignals** [comment]
@gajus I committed some changes that address your comments. Let me know if there's anything else. 

Thanks!

**Comment by gajus** [comment]
Noticed after merging that this was submitted without the accompanying documentation. That would be a nice follow up PR.

**Pull request merged**